lib/postgrex/type_server.ex

```diff
@@ -5,7 +5,7 @@ defmodule Postgrex.TypeServer do
 
   defstruct [:types, :connections, :lock, :waiting]
 
-  @timeout 60_000
+  @timeout Application.compile_env(:postgrex, :type_server_timeout, 60_000)
```

Comment:
I don't think it needs to be a compile time setting. Let's move it to runtime. Also, move the default value to the `def application` section in the `mix.exs`, this way we can use `fetch_env!` here. Thanks!